src/org/scratchOrgCreate.ts

```diff
@@ -98,11 +112,75 @@ const validateRetry = (retry: number): void => {
   }
 };
 
-const validateWait = (wait: Duration): void => {
-  const min = 2;
-  if (wait.minutes < min) {
-    throw messages.createError('WaitValidationMaxError', [min, wait.minutes]);
+export const scratchOrgResume = async (jobId: string): Promise<ScratchOrgCreateResult> => {
+  const [logger, cache] = await Promise.all([
+    Logger.child('scratchOrgResume'),
+    ScratchOrgCache.create(),
+    emit({ stage: 'send request' }),
+  ]);
+  logger.debug(`resuming scratch org creation for jobId: ${jobId}`);
+  if (!cache.has(jobId)) {
+    throw messages.createError('CacheMissError', [jobId]);
   }
+  const { hubUsername, apiVersion, clientSecret, signupTargetLoginUrlConfig, definitionjson, alias, setDefault } =
+    cache.get(jobId);
+
+  const hubOrg = await Org.create({ aliasOrUsername: hubUsername });
+  const soi = await queryScratchOrgInfo(hubOrg, jobId);
+
+  await validateScratchOrgInfoForResume({ jobId, soi, cache, hubUsername });
+  // At this point, the scratch org is "good".
+
+  // Some hubs have all the usernames set to `null`
+  const username = soi.Username ?? soi.SignupUsername;
+
+  // re-auth only if the org isn't in GlobalInfo
+  const globalInfo = await GlobalInfo.getInstance();
+  const scratchOrgAuthInfo = globalInfo.orgs.has(username)
+    ? await AuthInfo.create({
+        username,
+      })
+    : await authorizeScratchOrg({
+        scratchOrgInfoComplete: soi,
+        hubOrg,
+        clientSecret,
+        signupTargetLoginUrlConfig,
+        retry: 0,
+      });
+
+  const scratchOrg = await Org.create({ aliasOrUsername: username });
+
+  await emit({ stage: 'deploy settings', scratchOrgInfo: soi });
+  const settingsGenerator = new SettingsGenerator();
+  settingsGenerator.extract({ ...soi, ...definitionjson });
+  const [authInfo] = await Promise.all([
+    resolveUrl(scratchOrgAuthInfo),
+    deploySettings(
+      scratchOrg,
+      settingsGenerator,
+      apiVersion ??
+        (new ConfigAggregator().getPropertyValue('apiVersion') as string) ??
+        (await scratchOrg.retrieveMaxApiVersion())
+    ),
+  ]);
+
+  await scratchOrgAuthInfo.handleAliasAndDefaultSettings({
+    alias,
+    setDefault: setDefault ?? false,
+    setDefaultDevHub: false,
+  });
+  cache.unset(soi.Id ?? jobId);
+  const authFields = authInfo.getFields();
+
+  await Promise.all([emit({ stage: 'done', scratchOrgInfo: soi }), cache.write(), emitPostOrgCreate(authFields)]);
```

Comment:
the cache.unset above is also doing a write so the cache.write here is not needed
``` suggestion
  await Promise.all([emit({ stage: 'done', scratchOrgInfo: soi }), emitPostOrgCreate(authFields)]);
```